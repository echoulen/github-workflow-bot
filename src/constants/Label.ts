export enum Label {
  AWAIT_REVIEWER = "Awaiting reviewer feedback",
  AWAIT_AUTO_MERGE = "Awaiting auto merge",
  REVISION_NEEDED = "Revision needed",
  UPDATED = "updated",
  ONE_LGTM = "lgtm 1+",
}
