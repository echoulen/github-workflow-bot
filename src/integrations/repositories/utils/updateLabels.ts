import {List, Set} from "immutable";

export function updateLabels(labels: List<string>, addLabels: string[], removeLabels: string[]): string {
  return labels
    .concat(addLabels)
    .filter((label) => !Set(removeLabels).contains(label))
    .toSet()
    .join(",");
}
