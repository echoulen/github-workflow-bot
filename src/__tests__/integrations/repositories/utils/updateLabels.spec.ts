import {Label} from "@constants/Label";
import {updateLabels} from "@repositories/utils/updateLabels";

describe("updateLabels spec", () => {
  it("should updateLabels proper", () => {
    const result = updateLabels([Label.AWAIT_REVIEWER, Label.REVISION_NEEDED], [Label.UPDATED], [Label.REVISION_NEEDED]);
    expect(result).toMatchSnapshot();
  });
});
