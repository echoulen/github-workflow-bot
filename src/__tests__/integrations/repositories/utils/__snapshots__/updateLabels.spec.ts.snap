// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateLabels spec should updateLabels proper 1`] = `"Awaiting reviewer feedback,updated"`;
